### DISCUSSION TITLE
I made a tutorial on how to use nib files in DarwinKit

### DISCUSSION THREAD

**Discussion by programmingkidx** [progrium/darwinkit#248] [Show and tell]
Here is the tutorial: https://github.com/progrium/macdriver/wiki/How-to-use-nib-files-with-DarwinKit

Let me know what you think.

Thank you.

**Comment by Charliego3** [comment]
In this example get top-level objects
```go
var myObjects = f.Array_Array()
status := myNib.InstantiateWithOwnerTopLevelObjects(nil, unsafe.Pointer(&myObjects))
``` 

But the signature of the method is:
```go
func (n_ Nib) InstantiateWithOwnerTopLevelObjects(owner objc.IObject, topLevelObjects []objc.IObject) bool
```

Using this example `objects` is always empty
```go
//go:embed CollectionItem.nib
var collectionItemNib []byte
var objects = make([]objc.IObject, 0)
var nib = appkit.NewNibWithNibDataBundle(collectionItemNib, nil)
var status = nib.InstantiateWithOwnerTopLevelObjects(nil, objects)
```

**Reply by programmingkidx** [reply]
I think I know what is wrong. Recently there was a change to DarwinKit that changed how an Objective-C pointer to pointer type is translated to Go. Be sure you are using Darwinkit at git commit 6a98f82290e53bc5d82cb59436644f705156d7cb or higher. If you are using a git repo for DarwinKit try running 'git pull'. If you are not try downloading the latest version.

Thank you very much for the review.

**Reply by Charliego3** [reply]
Yes, I'm using latest version: [32e7360ca836dddaa9e3eed4659323e328eb9643](https://github.com/progrium/macdriver/commit/32e7360ca836dddaa9e3eed4659323e328eb9643)

https://github.com/progrium/macdriver/blob/32e7360ca836dddaa9e3eed4659323e328eb9643/macos/appkit/nib.gen.go#L89

**Reply by programmingkidx** [reply]
Could you try running this command: go generate ./macOS/appkit
It will recreate all the generated files for you.
The command is ran from the root level of your git repo.
